libs/langchain/langchain/schema/runnable/base.py

```diff
@@ -892,16 +882,18 @@ def batch(
 
         # invoke
         try:
-            for step in self.steps:
-                inputs = step.batch(
-                    inputs,
-                    [
-                        # each step a child run of the corresponding root run
-                        patch_config(config, rm.get_child())
-                        for rm, config in zip(run_managers, configs)
-                    ],
-                    max_concurrency=max_concurrency,
-                )
+            with get_executor_for_config(configs[0]) as executor:
```

Comment:
(not necessary) random thought -- we could consider adding an executor protocol and see if we can make async code look more symmetric to sync code by also using a context manager to yield an async executor